### ISSUE TITLE
Weapon blur being applied to 3rd person view

### ISSUE THREAD

**Issue by saxonschmauderer** [GarageGames/Torque3D#1997]
Issue:    While playing the game, there is a feature that blurs the gun when in the zoomed mode. This is supposed to provide a field of view perspective to make it look more realistic. Apart from the fact that it is kind of finicky and only works sometimes, it is also applied when it shouldn't be. 
When you are in the 3rd person camera angle and zoom in using the right mouse button, your character will be blurred. This looks awful and does not provide any benefits.

Proposed fix:    (1) Remove the applied blurring effect when in the 3rd person camera angle
                                                                                         or
                         (2) Disable the zoom feature while in 3rd person
                                                                                         or
                         (3) Have the zoom button change to first person and zoomed while the right mouse button is held down. This would provide the most functionality.

This is how the effect looks while in 1st person. (Actually boosts realism)
![blur 1st person](https://cloud.githubusercontent.com/assets/25248660/25504892/8dcb4f76-2b6d-11e7-8f1f-22868a00bf77.png)

This is how the effect looks while in 3rd person. (Degrades view and looks bad)
![blur 3rd person](https://cloud.githubusercontent.com/assets/25248660/25504937/c0cf633a-2b6d-11e7-97e8-ee3736f6f68d.png)

**Comment by Duion** [comment]
I disabled that effect alltogether since it does not work well overall, for example if you zoom on the sky it will apply max blur since it is infinitely far away, the problem is much more about that and not the 3rd person thing.
This is also just supposed to be a template and therefore you cannot make it right for everyone, disabling that feature is not that hard.

**Comment by Areloch** [comment]
Yeah, the DoF effect needs some updating, as Duion said for example, some of the default settings of it can go pretty oddball. It's largely designed with the default settings for first preson, but if someone wants it for third, it should probably at least handle it more gracefully than it does now with the default settings.

**Comment by saxonschmauderer** [comment]
I only propose bugs that I plan on fixing. I just want to put them out here for experienced users to see if they agree and mark as a bug or a feature request. I don't think the  default template should have the blur. I was thinking maybe we would switch to a first person zoomed perspective when aiming in the 3rd person view. Thoughts?

**Comment by Duion** [comment]
I don't think there are any settings you could use to make this problem go away.
There is also no difference in first person or third person, the postFX just does what it does, you can see the weapon blurred in the first image as well. 
PostFX are applied to the whole screen, they ignore what mode you are in, except you make a script to manually change the postFX in certain situations.

**Comment by Azaezel** [comment]
references for ya:

https://github.com/GarageGames/Torque3D/blob/development/Templates/Full/game/scripts/client/default.bind.cs#L451

https://github.com/GarageGames/Torque3D/blob/development/Templates/Full/game/scripts/client/default.bind.cs#L416-L418

**Comment by Areloch** [comment]
@saxonschmauderer Mulling on it, I think a good approach would be to have the zoom-in not introduce the blur, but instead we either shift it to being a default post effect that's got a options menu setting to override on/off, or we have a demoscene that utilizes it so people know how to use it. But it seems a bit odd for the view to go all DoF-y JUST when zoom occurs.

**Comment by Duion** [comment]
It makes sense as a concept, since when you focus on something the rest becomes more unfocussed and you can have DoF activated when not zooming as well, it will just get overwritten on zoom.
The only problem I have with this postFX is that if you aim at the sky the blur goes to the max value since the sky is infinitely far away, this is the ugly part.
But this all has not much to do with the original issue, that is not so much of an issue, since that is just the way it is supposed to work, if you don't want it that way you need a different shader.
